content/2015.09.kinto-encryption-example-fr.rst

```diff
@@ -0,0 +1,256 @@
+      };
+
+      return {
+        encode,
+        decode
+      };
+    };
+
+Nous créons deux instances Kinto, afin de pouvoir faire des tests de
+synchronization d'une instance à l'autre à l'aide du serveur de démo
+de Kinto. Créer plusieurs instances Kinto depuis la même origine est
+possible à l'aide de la toute récente option ``dbPrefix``:
+
+.. code-block:: JavaScript
+
+    // Kinto collection:
+    const createCollection = (transformer, testRun, instanceNo) => {
+      const kinto = new Kinto({
+        dbPrefix: `${testRun}-${instanceNo}`,
+        remote: 'https://kinto.dev.mozaws.net/v1/',
+        headers: {
+          Authorization: 'Basic ' + btoa('public-demo:s3cr3t')
+        }
+      });
+
+      return kinto.collection(`kinto-encryption-example-${testRun}`, {
+        remoteTransformers: [ transformer ]
+      });
+    };
+
+    var coll1, coll2;
+    const prepare = () => {
+      return generateAesKey().then(aesKey => {
+        return createTransformer(aesKey);
+      }).then(transformer => {
+        // Create two fresh empty Kinto instances for testing:
+        const testRun = new Date().getTime().toString();
+        coll1 = createCollection(transformer, testRun, 1);
+        coll2 = createCollection(transformer, testRun, 2);
+      });
+    };
+
+Maintenant, testons que nous puissions créer un nouvel enregistrement
+dans la collection 1, synchronisons le (les données envoyées au
+serveur doivent être chiffrées, ce qu'on peut vérifier avec l'onglet
+Réseau de la console du navigateur):
+
+.. image:: {filename}/images/kinto-encryption-example-network-tab.png
+   :alt: Onglet Réseau de la console Firefox
+
+.. code-block:: JavaScript
+
+    const syncUp = () => {
+      // Use first Kinto instance to demonstrate encryption:
+      return coll1.create({
+        URL: 'http://www.w3.org/TR/WebCryptoAPI/',
+        name: 'Web Cryptography API'
+      }).then(() => {
+        return coll1.sync();
+      }).then(syncResults => {
+        console.log('Sync up', syncResults);
+      });
+    };
+
+
+Puis synchronisons la collection 2 depuis le serveur Kinto. Là encore,
+les données téléchargées doivent être chiffrées mais les données
+déchiffrées doivent être stockées dans la base IndexDB et affichées
+dans le résultat de la synchronisation. Finalement, la fonction
+``go()`` permet de lancer tout le processus de test.
+
+Le code source complet est inclus dans la page que vous êtes
+actuellement en train de lire, alors n'hésitez pas, ouvrez la console
+du navigateur et essayez en lançant la fonction ``go()``!
+
+.. code-block:: JavaScript
+
+    const syncDown = () => {
+      // Use second Kinto instance to demonstrate decryption:
+      return coll2.sync().then(syncResults => {
+        console.log('Sync down', syncResults);
+      });
+    };
+
+    const go = () => {
+      console.log('Watch the Network tab!');
+      return prepare().then(() => {
+        return syncUp();
+      }).then(() => {
+        return syncDown();
```

Comment:
Not sure about that, but as `syncUp/syncDown` returns promises, I think the temporary arrow functions can be avoided here:

```
return prepare()
  .then(syncUp)
  .then(syncDown)
  .then(a => console.log('Success', a), b => console.error('Failure', b));
```